rskj-core/src/test/java/co/rsk/peg/BridgeSupportFlyoverTest.java

```diff
@@ -669,15 +670,19 @@ private BigInteger sendFundsSurpassesLockingCapToAnyAddress(
         BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
         when(provider.getNewFederation()).thenReturn(activeFederation);
         when(provider.getOldFederation()).thenReturn(retiringFederation);
+        when(provider.getLockingCap()).thenReturn(bridgeConstantsRegtest.getMaxRbtc());
+
         ReleaseTransactionSet releaseTransactionSet = new ReleaseTransactionSet(new HashSet<>());
         when(provider.getReleaseTransactionSet()).thenReturn(releaseTransactionSet);
 
         Address userRefundBtcAddress = PegTestUtils.createRandomBtcAddress(bridgeConstants.getBtcParams());
         Address lpBtcAddress = PegTestUtils.createRandomBtcAddress(bridgeConstants.getBtcParams());
 
-        Repository repository = spy(createRepository());
-        // Set current bridge balance in order to create the condition to surpass the locking cap when needed
-        doReturn(co.rsk.core.Coin.valueOf(1)).when(repository).getBalance(any());
+        Repository repository = createRepository();
+        // For simplicity of this test, the half of the max rbtc value is set as the current balance for the repository
+        // This means an amount sent over the half of the max rbtc value will surpass the locking cap which is set equal to that value
+        repository.addBalance(PrecompiledContracts.BRIDGE_ADDR, co.rsk.core.Coin.fromBitcoin(bridgeConstantsRegtest.getMaxRbtc().div(2)));
```

Comment:
To be honest, I find this solution a bit confusing. The idea is to test that the locking cap is exceeded but you are setting the locking cap as the max value. So it's actually like there's no locking cap.

I think the best solution is to receive the desired locking cap value as a parameter here, so each test can determine how much funds are included in the transaction and what the locking cap should be. That way you can determine on each test if the locking cap is exceeded or not.

We can discuss it further if you like